source/nodejs/adaptivecards-designer/src/card-designer-surface.ts

```diff
@@ -260,52 +260,52 @@ export class CardDesignerSurface {
     private renderCard() {
         this._cardHost.innerHTML = "";
 
-        if (this.card) {
-            if (this.onCardValidated) {
-                this.onCardValidated(this._lastParseErrors, this.card.validateProperties());
+        if (this.onCardValidated) {
+            let allValidationEvents: Adaptive.IValidationEvent[] = [];
+
+            for (let i = 0; i < this._serializationContext.eventCount; i++) {
+                allValidationEvents.push(this._serializationContext.getEventAt(i));
             }
 
-            let cardToRender: Adaptive.AdaptiveCard;
+            allValidationEvents.push(...this.card.validateProperties().validationEvents);
+
+            this.onCardValidated(allValidationEvents);
+        }
 
-            if (this.isPreviewMode) {
-                if (Shared.GlobalSettings.enableDataBindingSupport) {
-                    let cardPayload = this.card.toJSON();
+        let cardToRender: Adaptive.AdaptiveCard = this.card;
 
-                    try {
-                        let template = new ACData.Template(cardPayload);
-            
-                        let context = new ACData.EvaluationContext();
-                        context.$root = this.sampleData;
+        if (this.isPreviewMode) {
```

Comment:
Trying to understand this one - so today previewmode is designervnext is it? how are you choosing to define "preview" in the codebase today?

Reply:
This is unrelated to "preview features." The designer can operate in "design mode" and "preview mode"; this flag says whether the designer is in "preview mode."